### DISCUSSION TITLE
Konva Node Type (bug?)

### DISCUSSION THREAD

**Discussion by MentalGear** [konvajs/konva#1506] [Q&A] [answered]
I noticed that when running e.target.getType() on a Konva.Image, the output is 'Shape'  instead of 'Image'.

Is there any other way to detect if the user clicked on an image versus other nodes ?

**Comment by iaosee** [comment]
All shapes inherited from Konva.Shape output 'Shape',
You can use the `e.target instanceof Konva.Image` instead.

**Reply by MentalGear** [reply]
Thx, solved this already using getClassName, but your way is also interesting. Wondering though if it is intentional for getType() to always output `Shape` or if it is a bug.

**Reply by lavrton** [reply]
It is intentional.

**Comment by MentalGear** [comment, marked as answer]
Solved this by using
```js
node.getClassName() === "Image"
```
